.github/workflows/unit-tests.yaml

```diff
@@ -0,0 +1,200 @@
+#          - name: "Linux arm64"
+#            runner: matterlabs-ci-runner-arm
+#            target: "aarch64-unknown-linux-musl"
+#            rustflags: "-C link-arg=-Wl,-Bstatic -C link-arg=-lc -C link-arg=-L/usr/lib/llvm-15/lib/clang/15.0.7/lib/linux/ -C link-arg=-lclang_rt.builtins-aarch64"
+    runs-on: ${{ matrix.runner }}
+    name: ${{ matrix.name }}
+    container:
+      image: matterlabs/llvm_runner_jammy:latest
+      credentials:
+        username: ${{ secrets.DOCKERHUB_USER }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+    env:
+      RUSTFLAGS: ${{ matrix.rustflags }}
+      VYPER_DOWNLOAD_EXTENSION: ".linux"
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+      - name: Prepare environment
+        run: |
+          rustup target add ${{ matrix.target }}
+      - name: Install vyper compiler
+        run: |
+          curl --verbose \
+            --location -o "${PWD}/vyper" \
+            "${VYPER_DOWNLOAD_URL}/${VYPER_VERSION}/${VYPER_DOWNLOAD_FILENAME}${VYPER_DOWNLOAD_EXTENSION}" && \
+            chmod a+x "${PWD}/vyper"
+          echo "${PWD}" >> "${GITHUB_PATH}"
+      - name: Clone LLVM framework
+        run: |
+          cargo install compiler-llvm-builder --target ${{ matrix.target }}
+          zkevm-llvm clone
+      - name: Build LLVM framework
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 60
+          max_attempts: 2 # protection mechanism for MUSL sporadic download failure
+          command: zkevm-llvm build
+      - name: Build zkvyper
+        run: |
+          cargo build -vv --release --target ${{ matrix.target }}
+          echo "${PWD}/target/${{ matrix.target }}/release" >> "${GITHUB_PATH}"
+      - name: Run unit tests
+        run: |
+          cargo install cargo2junit
+          cargo test --verbose --target ${{ matrix.target }} -- -Z unstable-options --format json | cargo2junit > "${UNIT_TESTS_RESULTS_XML}"
+          cat "${UNIT_TESTS_RESULTS_XML}"
+      - name: Upload test results
+        if: always()
+        uses: EnricoMi/publish-unit-test-result-action@v2
+        with:
+          check_name: ${{ matrix.name }} Unit Tests Results
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          files: ${{ env.UNIT_TESTS_RESULTS_XML }}
+          action_fail_on_inconclusive: true
+
+  unit-tests-windows:
+    name: "Windows"
+    runs-on: windows-2022-github-hosted-16core
+    env:
+      LIBSTDCPP_SOURCE_PATH: "C:/a/_temp/msys64/mingw64/lib/libstdc++.a"
+      VYPER_DOWNLOAD_EXTENSION: ".windows.exe"
+    defaults:
+      run:
+        shell: msys2 {0}
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+      - name: Prepare msys2
+        uses: msys2/setup-msys2@v2
+      - name: Prepare env
+        run: |
+          pacman-key --refresh
+          pacman -Sy
+          curl -LO https://repo.msys2.org/mingw/mingw64/mingw-w64-x86_64-cmake-3.27.7-3-any.pkg.tar.zst
```

Comment:
Better style to have a long URL be in the env: section, than hard-coded in a command.

Reply:
Sure, agreed. I copied this code for Windows and did not pay too much attention to it. Nice that you marked that. Fixed.